### PR TITLE
WIP: Document ERR_load_<LIBRARY>_strings functions

### PR DESCRIPTION
**Pull request by richsalz**
This is in response to a recent comment on #13192 

It documents 39 functions, in a generic manner.

It is a WIP because if I leave the page in doc/internal/man3, I get errors like this:
```
ERR_load_ASN1_strings(3) is supposedly public but is documented as internal
```

But if I move it to doc/man3, I get errors like this:
```
crypto: function ERR_load_ASN1_strings(3) undocumented
```

I don't know how find-doc-nits works any more, so I don't know how to fix. More importantly, I don't know which fix is wanted -- are these functions public or not?

### CODE REVIEW

**Comment by paulidale** [comment]
We're hoping that these can be made private.  At least the new ones.  The existing ones should be deprecated.

**Comment by richsalz** [comment]
I strongly encourage you to treat them all the same way, whichever way it is, because you don't want to add flags to openssl.ec and special support to mkerr for something so, well, nit-picky.

I am happy to fix either error, if someone (@levitte?) can point me in the right direction.

**Comment by paulidale** [comment]
The line of thought was to generate internal functions and to manually produce the required wrappers for backwards compatibility.

I.e. treat them the same way after an initial cleansing.

**Comment by levitte** [comment]
> It is a WIP because if I leave the page in doc/internal/man3, I get errors like this:
> ```
> ERR_load_ASN1_strings(3) is supposedly public but is documented as internal
> ```

That happens when it finds documentation in `doc/internal/man3/`, but the symbols are still declared in public headers (i.e. those in `include/openssl/`)

> But if I move it to doc/man3, I get errors like this:
> ```
> crypto: function ERR_load_ASN1_strings(3) undocumented
> ```

Hmmm, no clue.  I'd have to see your worktree to know what went wrong...

File: doc/internal/man3/ERR_load_ASN1_strings.pod [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+ERR_load_UI_strings,
+ERR_load_X509V3_strings,
+ERR_load_X509_strings
+- load error message string table
+
+=head1 SYNOPSIS
+
+=for openssl generic
+
+int ERR_load_LIBRARY_strings(void)
```
Comment (mattcaswell) [inline doc/internal/man3/ERR_load_ASN1_strings.pod:49]: We should probably deprecate all these functions.

Comment (mattcaswell) [reply doc/internal/man3/ERR_load_ASN1_strings.pod:49]: (not necessarily in this PR though)

**Comment by richsalz** [comment]
I do not think there is anything unusual (I don't even know what a "worktree" is); you could just pull this PR and see if you get the same errors.

The project needs to decide if these are public or internal, first.

**Comment by richsalz** [comment]
still stuck waiting for input from the project.

**Comment by slontis** [comment]
> still stuck waiting for input from the project.

Not sure what is stuck here, the intention is that they all become internal at some point.
So the plan is to..
 - Deprecate the existing ones that are public.
 - Any that have been created during 3.0 should be internal.

@mattcaswell - does this need an OTC decision? (I don't think it does for these deprecation cases).

**Comment by mattcaswell** [comment]
> @mattcaswell - does this need an OTC decision? (I don't think it does for these deprecation cases).

I don't recall if this has ever been formally documented anywhere as our intended approach. If not an OTC decision might be wise.

**Comment by richsalz** [comment]
If you read all the comments on this PR, it is not clear that the project has come to a conclusion about what to do.  So this PR is waiting for the project to say something definitively.  If you read https://github.com/openssl/openssl/pull/13199#issue-507080416 then you will see that there is also an doc-nits issue; @levitte tried to help in https://github.com/openssl/openssl/pull/13199#issuecomment-713321542 but my reply seems to have gotten lost: what's a worktree?

**Comment by levitte** [comment]
> what's a worktree?

[gitglossary(7)](https://man7.org/linux/man-pages/man7/gitglossary.7.html) calls it "working tree"...  I've used "worktree", which is imprecise in the most literal sense, but I hoped it was close enough to be comprehensible.

**Comment by levitte** [comment]
An idea that was floated in a recent OTC meeting was to collect all the ERR_load_{LIBRARY}_strings() functions that exist in 1.1.1 into a compatibility header file, make them all deprecated, and stop having util/mkerr.h add their declaration in the generated {library}err.h headers.  That way, util/mknum.pl wouldn't add util/libcrypto.num entries for those added by post-1.1.1 code, and find-doc-nits would have no reason to complain about the lack of documentation for them.

**Comment by richsalz** [comment]
So the project is still figuring out what to do; that's fine.  As for worktree, there is none.  Just clone the PR and try the experiment.

**Comment by levitte** [comment]
> So the project is still figuring out what to do; that's fine.

It's an idea on how to implement it.  Since this is something this PR (*your* PR) is dealing with, that may be an idea for you to pick up on and try out...  of course, if you're willing.  If not, someone else will pick it up sooner or later and implement it, probably in a separate PR.  You tell us how you want to go forward from here.

**Comment by richsalz** [comment]
It would be nice to know how to fix the issues I described in the first comment on this PR. Then, when the project figures out what to do, implementation is easier.

I was just trying to help make 3.0 happen quicker.

**Comment by levitte** [comment]
Ok, I'm apparently not getting through to you...

**Comment by levitte** [comment]
#13390 will hopefully resolve the questions asked here.

**Comment by richsalz** [comment]
Took a brief look at #13390 and it seems this PR should be closed, so I'm doing that.

**Comment by levitte** [comment]
> Took a brief look at #13390 and it seems this PR should be closed, so I'm doing that.

Er....  #13390 does *nothing* in terms of documentation, apart from removing ERR_load functions that are master only...  so I'm not sure what gave you the idea that PR overrides this one.

**Pull request closed without merging**